src/com/google/cose/EncryptionKey.java

```diff
@@ -139,4 +148,36 @@ public static EncryptionKey parse(byte[] keyBytes) throws CborException, CoseExc
   public static EncryptionKey decode(DataItem cborKey) throws CborException, CoseException {
     return new EncryptionKey(cborKey);
   }
+
+  public byte[] encrypt(Algorithm algorithm, byte[] message, byte[] iv, byte[] aad)
+      throws CoseException {
+    try {
+      Cipher cipher = Cipher.getInstance("AES/GCM/NoPadding");
+      GCMParameterSpec gcmSpec = new GCMParameterSpec(16 * 8, iv);
+      cipher.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(secretKey, algorithm.getJavaAlgorithmId()),
+          gcmSpec);
+      if (aad != null) {
+        cipher.updateAAD(aad);
+      }
+      return cipher.doFinal(message);
+    } catch (NoSuchAlgorithmException | NoSuchPaddingException | InvalidAlgorithmParameterException
+        | InvalidKeyException | IllegalBlockSizeException | BadPaddingException e) {
+      throw new CoseException("Error while encrypting message.", e);
+    }
+  }
+
+  public byte[] decrypt(Algorithm algorithm, byte[] message, byte[] iv, byte[] aad)
```

Comment:
I moved out the bulk of the functionality to a separate function which accepts a `Cipher` mode. However, I am still keeping these functions around since I want the users to have `encrypt` and `decrypt` functions explicitly that they can call